core/tests/app/test_app.py

```diff
@@ -505,17 +561,174 @@ def test_set_empty_full_screen_window_list(event_loop):
+    app.enter_presentation_mode({})
+    assert not app.is_presentation_mode
+    assert_action_not_performed(app, "enter presentation mode")
+
+    # Enter presentation mode with an 1 element screen-window dict:
+    app.enter_presentation_mode({app.screens[0]: window1})
+    assert app.is_presentation_mode
+    assert_action_performed_with(
+        app,
+        "enter presentation mode",
+        screen_window_dict={app.screens[0]: window1},
+    )
+    # Exit presentation mode:
+    app.exit_presentation_mode()
+    assert_action_performed(
+        app,
+        "exit presentation mode",
+    )
+
+    # Enter presentation mode with a 2 elements screen-window dict:
+    app.enter_presentation_mode({app.screens[0]: window1, app.screens[1]: window2})
+    assert app.is_presentation_mode
+    assert_action_performed_with(
+        app,
+        "enter presentation mode",
+        screen_window_dict={app.screens[0]: window1, app.screens[1]: window2},
+    )
+    # Exit presentation mode:
+    app.exit_presentation_mode()
+    assert_action_performed(
+        app,
+        "exit presentation mode",
+    )
+
+
+def test_presentation_mode_with_excess_windows_list(event_loop):
+    """Entering presentation mode limits windows to available displays."""
+    app = toga.App(formal_name="Test App", app_id="org.example.test")
+    window1 = toga.Window()
+    window2 = toga.Window()
+
+    assert not app.is_presentation_mode
+
+    # Entering presentation mode with 3 windows should drop the last window,
+    # as the app has only 2 screens:
+    app.enter_presentation_mode([app.main_window, window2, window1])
+    assert app.is_presentation_mode
+    assert_action_performed_with(
+        app,
+        "enter presentation mode",
+        screen_window_dict={app.screens[0]: app.main_window, app.screens[1]: window2},
+    )
+    # Exit presentation mode:
+    app.exit_presentation_mode()
+    assert_action_performed(
+        app,
+        "exit presentation mode",
+    )
+
+
+def test_presentation_mode_no_op(event_loop):
+    """Entering presentation mode with invalid conditions is a no-op."""
+    app = toga.App(formal_name="Test App", app_id="org.example.test")
+
+    assert not app.is_presentation_mode
+
+    # If we're not in presentation mode, exiting presentation mode is a no-op.
+    app.exit_presentation_mode()
+    assert_action_not_performed(app, "exit presentation mode")
+
+    # Entering presentation mode without any window is a no-op.
+    with pytest.raises(TypeError):
+        app.enter_presentation_mode()
+        assert_action_not_performed(app, "enter presentation mode")
+
+    # Entering presentation mode without proper type of parameter is a no-op.
+    with pytest.raises(
+        TypeError,
+        match="Invalid type for windows parameter. Expected list or dict type.",
+    ):
+        app.enter_presentation_mode(app.main_window)
+        assert_action_not_performed(app, "enter presentation mode")
```

Comment:
Since, the state checks are handled by the backend, the reason being:
https://github.com/beeware/toga/blob/7cd1489f525244ee8cdb68067994037bf56ad1d1/core/src/toga/window.py#L489-L494

Hence, tests are for entering presentation mode if we are already in full screen mode are being done at the testbed test: `tests/window/test_window.py::test_window_state_direct_change_with_intermediate_states`